src/libraries/System.Text.Json/src/CompatibilitySuppressions.xml

```diff
@@ -57,6 +57,54 @@
     <Right>lib/netstandard2.0/System.Text.Json.dll</Right>
     <IsBaselineSuppression>true</IsBaselineSuppression>
   </Suppression>
+  <Suppression>
```

Comment:
Explicitly adding default constructors tripped off the API compat checker necessitating these suppressions. FYI @carlossanlop @ViktorHofer

Reply:
That needs to be investigated. @eiriktsarpalis do you still have the error message available? I wonder if APICompat complains about the member being on the ref but not on the src or the other way around.

cc @ericstj as I'm out the next two weeks

Reply:
Actually, I just looked at the reference source and the type defines an explicit constructor with parameters, so adding a default one is an API addition that you probably don't want to make. Or do you?

Reply:
Yeah these are all ref/lib comparisons.  

> Explicitly adding default constructors

This isn't a case of explicit vs implicit.  Implicit default constructors [only exist if no other constructors are defined](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/instance-constructors#parameterless-constructors). In these cases you're adding a default constructor to types that didn't have an implicit one.  This a real error. 
<img width="363" alt="image" src="https://github.com/dotnet/runtime/assets/8918108/aabc0de0-48a1-4efe-a30d-b90c2eabe74f">

If the API docs tooling is telling you this constructor exists -- it is mistaken.  Don't add this API.

Reply:
> Yeah these are all ref/lib comparisons.

That's correct. I was asking about the error message as it provides information about if the error applies to the right or the left side. As this comparison is done in strict mode, the CP0002 error could apply to either missing API on the left or on the right side. Unfortunately, the XML doesn't provide that detail.

That said, in this case it's clear that this is about an API addition on the right side.

Reply:
Thanks. The API docs tooling is actually correct -- it was complaining about a missing parameter that I mistook for a missing default constructor (similar to a few other types in the undoc list). I just reverted the unnecessary constructors and suppressions.